avm/res/event-grid/system-topic/event-subscription/main.bicep

```
@@ -7,17 +7,84 @@ param name string
 @description('Required. Name of the Event Grid System Topic.')
 param systemTopicName string
 
-@description('Optional. Dead Letter Destination. (See https://learn.microsoft.com/en-us/azure/templates/microsoft.eventgrid/eventsubscriptions?pivots=deployment-language-bicep#deadletterdestination-objects for more information).')
-param deadLetterDestination object?
+// User-defined types for Event Grid components  
+@description('Event subscription destination configuration. Should contain endpointType and properties matching the Azure Event Grid destination schema.')
+type destinationType = object
+
+@description('Dead letter destination configuration. Should contain endpointType and properties matching the Azure Event Grid dead letter destination schema.')
+type deadLetterDestinationType = object
+
+@description('Identity configuration for delivery or dead letter.')
+type identityType = {
+  @description('Required. The type of identity to use.')
+  type: 'SystemAssigned' | 'UserAssigned'
+  
+  @description('Conditional. Required if type is UserAssigned. The user assigned identity resource ID.')
+  userAssignedIdentity: string?
+}
+
+@description('Delivery configuration with resource identity.')
+type deliveryWithResourceIdentityType = {
```

```suggestion
export()
type deliveryWithResourceIdentityType = {
```

Allows consumers of the module to reuse the type